qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py

```diff
@@ -0,0 +1,256 @@
+            for start, end, weight in weighted_edges:
+                graph.update_edge(start, end, weight)
+        self._graph = graph
+
+        self.pos: Optional[dict] = None
+
+    @property
+    def graph(self) -> PyGraph:
+        """Return a copy of the input graph."""
+        return self._graph.copy()
+
+    @property
+    def num_nodes(self) -> int:
+        """Return the number of nodes."""
+        return self.graph.num_nodes()
+
+    @property
+    def node_indexes(self) -> NodeIndices:
+        """Return the node indexes."""
+        return self.graph.node_indexes()
+
+    @property
+    def weighted_edge_list(self) -> WeightedEdgeList:
+        """Return a list of weighted edges."""
+        return self.graph.weighted_edge_list()
+
+    def copy(self) -> "Lattice":
+        """Return a copy of the lattice."""
+        return Lattice(self.graph.copy())
+
+    @classmethod
+    def from_nodes_and_edges(
+        cls, num_nodes: int, weighted_edges: List[Tuple[int, int, complex]]
+    ) -> "Lattice":
+        """Return an instance of Lattice from the number of nodes and the list of edges.
+
+        Args:
+            num_nodes: The number of nodes.
+            weighted_edges: A list of tuples consisting of two nodes and the weight between them.
+        Returns:
+            Lattice generated from lists of nodes and edges.
+        """
+        graph = PyGraph(multigraph=False)
+        graph.add_nodes_from(range(num_nodes))
+        graph.add_edges_from(weighted_edges)
+        return cls(graph)
+
+    def to_adjacency_matrix(self, weighted: bool = False) -> np.ndarray:
+        """Return its adjacency matrix from weighted edges.
+        The weighted edge list is interpreted as the upper triangular matrix.
+        Defaults to False.
+
+        Args:
+            weighted: The matrix elements are 0 or 1 when it is False.
+                Otherwise, the weights on edges are returned as the matrix elements.
+
+        Returns:
+            The adjacency matrix of the input graph.
+        """
+        if weighted:
+            real_part = adjacency_matrix(self.graph, weight_fn=np.real)
+            imag_part = adjacency_matrix(self.graph, weight_fn=np.imag)
+            imag_part = np.triu(imag_part) - np.triu(imag_part).T
+            ad_mat = real_part + 1.0j * imag_part
+
+        else:
+            ad_mat = adjacency_matrix(self.graph, weight_fn=lambda x: 1)
+
+        return ad_mat
+
+    @staticmethod
+    def _mpl(graph: PyGraph, self_loop: bool, **kwargs):
+        """
+        Auxiliary function for drawing the lattice using matplotlib.
+
+        Args:
+            graph : graph to be drawn.
+            self_loop : Draw self-loops, which are edges connecting a node to itself.
+            **kwargs : Kwargs for drawing the lattice.
+
+        Returns:
+            A matplotlib figure for the visualization if not running with an
+            interactive backend (like in jupyter) or if ``ax`` is not set.
```

Comment:
I think again this does not return anything. Since its a private method it will not end up in our published docs but maybe it should be corrected anyway.